relay-cabi/src/processing.rs

```diff
@@ -321,3 +325,72 @@ pub unsafe extern "C" fn relay_validate_project_config(
         Err(e) => RelayStr::from_string(e.to_string()),
     }
 }
+
+#[derive(Debug, Deserialize)]
+struct EphemeralEvent {
+    transaction: String,
+}
+
+impl EphemeralEvent {
+    fn to_event(&self) -> Event {
+        Event {
+            transaction: Annotated::new(self.transaction.clone()),
+            ty: Annotated::new(EventType::Transaction),
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct EphemeralSamplingResult {
+    merged_sampling_configs: Vec<SamplingRule>,
+    sampling_match: Option<SamplingMatch>,
+}
+
+/// Runs dynamic sampling given the sampling config, root sampling config, dsc and event.
+///
+/// Returns the sampling decision containing the sample_rate and the list of matched rule ids.
+#[no_mangle]
+#[relay_ffi::catch_unwind]
+pub unsafe extern "C" fn run_dynamic_sampling(
+    sampling_config: *const RelayStr,
+    root_sampling_config: *const RelayStr,
+    dsc: *const RelayStr,
+    event: *const RelayStr,
```

Comment:
You can pass all of these as references and then don't have to worry about dereferencing the pointer below. This is something we've not done consistently in the codebase, but we should move towards:

```suggestion
    sampling_config: &RelayStr,
    root_sampling_config: &RelayStr,
    dsc: &RelayStr,
    event: &RelayStr,
```